annales-medico-psychologiques.csl

```diff
@@ -0,0 +1,153 @@
+  <info>
+    <title>Annales médico-psychologiques</title>
+    <id>http://www.zotero.org/styles/annales-medico-psychologiques</id>
+    <link href="http://www.zotero.org/styles/annales-medico-psychologiques" rel="self"/>
+    <link href="https://www.em-consulte.com/getInfoProduit/AMEPSY/instructionsAuteurs/AMEPSY.pdf" rel="documentation"/>
+    <author>
+      <name>Richard Karnesky</name>
+      <email>karnesky+zotero@gmail.com</email>
+      <uri>http://arc.nucapt.northwestern.edu/Richard_Karnesky</uri>
+    </author>
+    <contributor>
+        <name>Sylvain Machefert</name>
+        <email>zotero@geobib.fr</email>
+        <uri>https://orcid.org/0000-0001-9288-1754</uri>
+    </contributor>
+    <category citation-format="numeric"/>
+    <issn>0003-4487</issn>
+    <summary>A style for Annales médico-psychologiques, based on elsevier-vancouver but with alphabetical sorting of biblio as specified in the instructions to authors.</summary>
+    <updated>2022-03-30T09:23:38+00:00</updated>
+    <rights license="http://creativecommons.org/licenses/by-sa/3.0/">This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 License</rights>
+  </info>
+  <macro name="author">
+    <names variable="author">
+      <name initialize-with="" delimiter=", " delimiter-precedes-last="always" name-as-sort-order="all" sort-separator=" "/>
+      <label form="long" prefix=", "/>
+      <substitute>
+        <names variable="editor"/>
+        <names variable="translator"/>
+      </substitute>
+    </names>
+  </macro>
+  <macro name="editor">
+    <names variable="editor">
+      <name initialize-with="" delimiter=", " delimiter-precedes-last="always" name-as-sort-order="all" sort-separator=" "/>
+      <label form="long" prefix=", " suffix="."/>
+    </names>
+  </macro>
+  <macro name="year-date">
+    <choose>
+      <if variable="issued">
+        <date variable="issued">
+          <date-part name="year"/>
+        </date>
+      </if>
+      <else>
+        <text term="no date" form="short"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="publisher">
+    <text variable="publisher-place" suffix=": " text-case="title"/>
+    <text variable="publisher" suffix="; "/>
+    <text macro="year-date"/>
+  </macro>
+  <macro name="edition">
+    <choose>
+      <if is-numeric="edition">
+        <group delimiter=" ">
+          <number variable="edition" form="ordinal"/>
+          <text term="edition" form="short"/>
+        </group>
+      </if>
+      <else>
+        <text variable="edition"/>
+      </else>
+    </choose>
+  </macro>
+  <citation collapse="citation-number">
+    <sort>
+      <key variable="citation-number"/>
+    </sort>
+    <layout prefix="[" suffix="]" delimiter=",">
+      <text variable="citation-number"/>
+    </layout>
+  </citation>
+  <bibliography entry-spacing="0" second-field-align="flush" et-al-min="7" et-al-use-first="6">
+    <sort>
+      <key variable="author" sort="ascending"/>
```

Comment:
I think you should add a second sort key if the author key is identical. (Probably using `title`?)

Reply:
Also, check how references without authors are sorted.